src/fluree/db/query/fql.cljc

```diff
@@ -618,16 +618,21 @@
                           res)]
                (if inVector? [res'] res'))
 
-             (let [res+agg (if aggregates (analytical/add-aggregate-cols res aggregates) res)
-                   offset  (if groupBy 0 offset)
-                   {:keys [headers tuples]} (if orderBy
-                                              (order-offset-and-limit-results orderBy res+agg offset group-limit)
-                                              res+agg)
-                   res     (<? (format-filter-tuples db tuples select-spec headers vars (dissoc opts :limit :offset :orderBy :groupBy)))]
+             (let [{:keys [headers tuples]} (if aggregates (analytical/add-aggregate-cols res aggregates) res)
+                   offset' (if (or selectDistinct? groupBy) 0 offset)
+                   limit'  (if selectDistinct? nil group-limit)
+                   tuples' (if orderBy
+                             (order-offset-and-limit-results tuples headers orderBy offset' limit')
+                             tuples)
+                   res'    (->> (dissoc opts :limit :offset :orderBy :groupBy)
+                                (format-filter-tuples db tuples' select-spec headers vars)
+                               <?)]
```

Comment:
Indentation needs one more scoot to the right. :)

Reply:
thanks - noted & will make change as I start to refactor a bit